install/install_test.go

```diff
@@ -0,0 +1,71 @@
+package install
+
+import (
+	"gopkg.in/check.v1"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (b *InstallSuite) TestTail(c *check.C) {
+	cases := []struct {
+		given  []*corev1.Pod
+		expect []*corev1.Pod
+	}{
+		{
+			given: []*corev1.Pod{
+				{ObjectMeta: v1.ObjectMeta{Name: "Foo"}}},
+			expect: nil,
+		},
+		{
+			given: []*corev1.Pod{
+				{ObjectMeta: v1.ObjectMeta{Name: "Foo"}},
+				{ObjectMeta: v1.ObjectMeta{Name: "Bar"}}},
+			expect: []*corev1.Pod{{ObjectMeta: v1.ObjectMeta{Name: "Bar"}}},
+		},
+		{
+			given:  []*corev1.Pod{},
+			expect: nil,
+		},
+		{
+			given: []*corev1.Pod{
+				{ObjectMeta: v1.ObjectMeta{Name: "Foo"}},
+				{ObjectMeta: v1.ObjectMeta{Name: "Bar"}},
+				{ObjectMeta: v1.ObjectMeta{Name: "Qux"}}},
+			expect: []*corev1.Pod{{ObjectMeta: v1.ObjectMeta{Name: "Bar"}}, {ObjectMeta: v1.ObjectMeta{Name: "Qux"}}},
+		},
+	}
+	for _, cs := range cases {
+		got := tail(cs.given)
+		c.Assert(got, check.DeepEquals, cs.expect, check.Commentf("Got %v, expected %v", got, cs.expect))
+	}
+}
+
+func (b *InstallSuite) TestGroupPodsByNamespace(c *check.C) {
+	t := corev1.PodList{
+		Items: []corev1.Pod{
+			{ObjectMeta: v1.ObjectMeta{Name: "Foo", Namespace: "Berlin"}},
+			{ObjectMeta: v1.ObjectMeta{Name: "Bar", Namespace: "Berlin"}},
+			{ObjectMeta: v1.ObjectMeta{Name: "Qux", Namespace: "Caracas"}},
+		},
+	}
+
+	nsToPods := groupPodsByNamespace(&t)
+	c.Assert(nsToPods["Berlin"], check.DeepEquals, []*corev1.Pod{{ObjectMeta: v1.ObjectMeta{Name: "Foo", Namespace: "Berlin"}},
+		{ObjectMeta: v1.ObjectMeta{Name: "Bar", Namespace: "Berlin"}}})
+	c.Assert(nsToPods["Caracas"], check.DeepEquals, []*corev1.Pod{{ObjectMeta: v1.ObjectMeta{Name: "Qux", Namespace: "Caracas"}}})
+}
+
+func (b *InstallSuite) TestRunEvictions(c *check.C) {
+	// it should not evict cep pods
```

Comment:
Are you planning on implementing this?

Reply:
I will remove this. I discussed this with @kaworu , and it seems that this kind of test can be done on the CI, e.g.: asserting that evictions happen as expected. This will make the whole thing easier, otherwise we would need to mock too many things for the value that we get.